### PR TITLE
Airship-3071: Migrate to github actions

### PR DESCRIPTION
**Pull request by mmcgarr**

### CODE REVIEW

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: push
+on: push
+permissions:
+    contents: read
+    id-token: write
```
Comment (DewaldV) [inline .github/workflows/push.yaml]: Do we need an ID token for this? I don't think we're using it in this.

**Review by meghaniankov — DISMISSED** [review]

File: .github/workflows/push.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        images: quay.io/uswitch/vault-creds
+        tags: |
+          type=semver,pattern={{version}}
+          type=sha,prefix=,format=long,
+    - uses: docker/build-push-action@v4
+      with:
+        context: .
+        labels: ${{ steps.meta.outputs.labels }}
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
```
Comment (DewaldV) [inline .github/workflows/push.yaml:54]: I don't see a `docker login` step, how do we auth with `quay.io`?

File: .github/workflows/push.yaml (hunk 1 above)
Comment (mmcgarr) [reply .github/workflows/push.yaml]: thanks, removed it 👍

File: .github/workflows/push.yaml (hunk 2 above)
Comment (mmcgarr) [reply .github/workflows/push.yaml:54]: Good spot, added the login step which I forgot

**Review by DewaldV — APPROVED** [review]

**Pull request merged**